lib/generate-svg.js

```diff
@@ -72,8 +72,22 @@ module.exports = (files = [], options = {}) => {
             svg.setAttribute('xmlns:xlink', 'http://www.w3.org/1999/xlink');
         }
 
+        //get prefix type to deal with functions
+        const prefixType = typeof options.sprite.prefix;
+        if (prefixType === 'function')
+            var prefixFunc = options.sprite.prefix;
+
         Promise.all(files.map((file) => new Promise((resolve) => {
+            if (options.sprite.prefix) {
+                if (prefixType === 'function') {
+                    options.sprite.prefix = prefixFunc(file);
+                } else {
+                    options.sprite.prefix = options.sprite.prefix;
+                }
+            }
```

Comment:
`generateStyles()` uses the prefix as well and this wouldn't work as they sould match, see https://github.com/cascornelissen/svg-spritemap-webpack-plugin/blob/508fe382fee6e8b3a9971cfb0d67f68be9c29dbb/lib/index.js#L138

Could you move this to a helper (e.g. `/lib/helpers/generate-sprite-prefix.js`) function and call it in all required places? Something like

```es6
module.exports = (prefix, file) => {
    if ( typeof prefix === 'function' ) {
        return prefix(file);
    }    

    return prefix;
}
```

And then use it here as

```
const prefix = generateSpritePrefix(options.sprite.prefix, file);
```

And then use `prefix` instead of `options.sprite.prefix` in the rest of this block.